lib/sles4sap.pm

```diff
@@ -1463,6 +1511,45 @@ sub get_instance_profile_path () {
     return ($profile_path);
 }
 
+=head2 load_ase_env
+
+  $self->load_ase_env
+
+Loads environment variables from ASE installation into the current shell session.
+
+=cut
+
+sub load_ase_env {
+    my ($self) = @_;
+    return unless $self->ASE_RESPONSE_FILE;
+
+    # SAP ASE installation leaves a SYBASE.sh file with environment variables definitions in
+    # the directory where it was installed. The command below will extract the value of the
+    # install directory from the response file and prepend it to SYBASE.sh to load those
+    # variables
+    # Command will look like this:
+    # source $(awk -F= '/^USER_INSTALL_DIR/ {print $2}' $HOME/ASE_RESPONSE_FILE.txt)/SYBASE.sh
+    assert_script_run q|source $(awk -F= '/^USER_INSTALL_DIR/ {print $2}' $HOME/| . $self->ASE_RESPONSE_FILE . ')/SYBASE.sh';
```

Comment:
Using the quote operator to do a regex inside the SUT and concatenate that again in Perl looks a bit write only to me.

Reply:
I don't follow. Quote operator is used to keep in the code something that is more readable. I'm open to suggestions on how to write this and make it clearer, because I'm not happy with the current solution (and that's the reason for the comment above the line).

So, in short, the command we want to send to SUT is:
```
source $(awk -F= '/^USER_INSTALL_DIR/ {print $2}' $HOME/ASE_RESPONSE_FILE.txt)/SYBASE.sh
```
This was originally written in Perl as:
```
assert_script_run q|source $(awk -F= '/^USER_INSTALL_DIR/ {print $2}' $HOME/ASE_RESPONSE_FILE.txt)/SYBASE.sh|;
```
Using the quote operator so both the actual command in shell and in Perl would look the same. Single quotes were not used as they would clash with the single quotes in the `awk` command, and double quotes were not used as it would require to escape the `$` signs in the shell command.

However, after it was pointed out that this was hardcoding the name of the response file, the command was modified first by using a variable:
```
assert_script_run "source \$(awk -F= '/^USER_INSTALL_DIR/ {print \$2}' \$HOME/$response_file)/SYBASE.sh";
```
And then by using a class property, `$sles4sap::ASE_RESPONSE_FILE`. Since the class property works similar to a function, the command with the double quotes could not work verbatim. It had to be changed to:
```
assert_script_run "source \$(awk -F= '/^USER_INSTALL_DIR/ {print \$2}' \$HOME/" . $self->ASE_RESPONSE_FILE . '/SYBASE.sh';
```
But - again - with double quotes and no Perl variable in the string, I feel the left side of the command is not as readable than with the quote operator.

One option would be to use an aux Perl variable and keep the double quotes, for example:
```
my $response_file = $self->ASE_RESPONSE_FILE;
assert_script_run "source \$(awk -F= '/^USER_INSTALL_DIR/ {print \$2}' \$HOME/$response_file)/SYBASE.sh";
```
I can change it to that if you feel it's more clear, or to any other suggestion which improves readability.

Reply:
I would use perl to do `awk -F= '/^USER_INSTALL_DIR/ {print \$2}'` because I think we should minimize the amount of languages we are using in our tests.